handler/src/main/java/io/netty/handler/stream/ChunkedWriteHandler.java

```diff
@@ -68,24 +68,23 @@
 public class ChunkedWriteHandler extends ChannelDuplexHandler {
 
     private static final InternalLogger logger =
-        InternalLoggerFactory.getInstance(ChunkedWriteHandler.class);
+            InternalLoggerFactory.getInstance(ChunkedWriteHandler.class);
 
     private final Queue<PendingWrite> queue = new ArrayDeque<PendingWrite>();
```

Comment:
I think the correct fix is to just use `PendingWriteQueue` as this one will correctly update `Channel.isWritable()`.

Reply:
Thanks for the fast feedback, let me change it :+1:

Reply:
@normanmaurer OT, I would like to implement a version of `PendingWriteQueue` as an [UnrolledLinkedList](https://en.wikipedia.org/wiki/Unrolled_linked_list) to reduce footprint and improve cache friendliness wdyt?

Reply:
sure lets try this and add a benchmark :)

Reply:
I've pushed the change, trying to minimize the impact on the current behaviour and re previous fixes, but probably rewriting the logic of the handler would be more beneficial, long term.

I see that `doFlush` will continue reading from the queue if `!isWritable`: assuming a relevant backlog of pending writes into a `queue` and a !writable channel (due to it), how the handler is supposed to continue draining from the `queue`?

Reply:
I've pushed a new change that disable writability checks on doFlush: doFlush always take care of flushing the last written data and it would save pending `ByteBuf`s into `queue` causing !wriable channel to not being flushed anymore, making the writability to became `true`. I've yet to understand if is the right way to handle this case.